src/evm/protocol/uniswap_v3/state.rs

```diff
@@ -458,6 +562,54 @@ impl ProtocolSim for UniswapV3State {
         Ok(())
     }
 
+    fn swap_to_price(
+        &self,
+        token_in: &Bytes,
+        token_out: &Bytes,
+        target_price: Price,
+    ) -> Result<BigUint, SimulationError> {
+        match self.execute_swap_to_price_limit(token_in, token_out, &target_price)? {
+            Some((amount_in, _, _)) => Ok(u256_to_biguint(amount_in)),
+            None => Ok(BigUint::zero()),
+        }
+    }
```

Comment:
Please add some docs on how this works. It will be simpler than in uni-v2. you have an example docs from APEX